ports/rp2/main.c

```diff
@@ -199,6 +208,10 @@ int main(int argc, char **argv) {
 
     soft_reset_exit:
         mp_printf(MP_PYTHON_PRINTER, "MPY: soft reboot\n");
+
+        // Hook for resetting anything immediately following a soft reboot command
+        MICROPY_BOARD_EARLY_RESET();
```

Comment:
As above, a similar hook in the stm32 and renesas-ra ports is `MICROPY_BOARD_START_SOFT_RESET(&state)`. Although these hooks have default values on these ports (and are in a slightly different position), would it be possible to make this hook the same as stm32/renesas for consistency?

Reply:
This one I actively use, but I am happy to change it.

The different line seems to be because it is called before the `mp_printf` on STM, which I think is the better approach, as that state also gives the option to disable the print if a build requires.